### ISSUE TITLE
Unclear warning message

### ISSUE THREAD

**Issue by L42y** [smyte/jsxstyle#78]
```
WARNING in ./components/Admin/Table/Column.js
    `component` prop value was not handled by extractStyles: isSelf ? Link : 'a'
```
since `component` prop won't be extract as style, I assume this warning message is not necessary?

**Comment by meyer** [comment]
@L42y ah yes. That’s a poorly-worded warning message, but it is a valid one. The AST transform only handles the following two cases for the value of the `component` prop:

```jsx
<Block component={Thing} />
<Block component="div" />
```

The way to get around that warning for now is to do the following:

```jsx
const LinkComponent = isSelf ? Link : 'a';

<Block component={LinkComponent} />
```

In the future I can automatically generate an uppercase component name if the value of the `component` prop isn’t a string literal (`"div"`) or an identifier with an uppercase first letter (`Thing`). We have also encountered this warning message in a few places in the Smyte codebase.

**Comment by meyer** [comment]
@L42y I’ve added a transform for complex `component` prop values. Let me know if it works for you. You’ll still get a warning, but it’ll look like this:

```
Complex `component` prop value (`isSelf ? Link : 'a'`) will be extracted out as a separate variable declaration.
```

**Comment by meyer** [comment]
published as `jsxstyle-loader@1.0.0-rc.4`.
